entity-framework/core/modeling/value-conversions.md

```diff
@@ -2,145 +2,746 @@
 
-Note that all the built-in converters are stateless and so a single instance can be safely shared by multiple properties.
+<!--
+        public class Post
+        {
+            public int Id { get; set; }
+            public string Title { get; set; }
+            public string Contents { get; set; }
 
-## Pre-defined conversions
+            public ICollection<string> Tags { get; set; }
+        }
+-->
+[!code-csharp[PrimitiveCollectionModel](../../../samples/core/Modeling/ValueConversions/PrimitiveCollection.cs?name=PrimitiveCollectionModel)]
 
-For common conversions for which a built-in converter exists there is no need to specify the converter explicitly. Instead, just configure which provider type should be used and EF will automatically use the appropriate built-in converter. Enum to string conversions are used as an example above, but EF will actually do this automatically if the provider type is configured:
+Using <xref:System.Text.Json> again:
 
 ```csharp
-modelBuilder
-    .Entity<Rider>()
-    .Property(e => e.Mount)
-    .HasConversion<string>();
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder.Entity<Post>()
+                    .Property(e => e.Tags)
+                    .HasConversion(
+                        v => JsonSerializer.Serialize(v, null),
+                        v => JsonSerializer.Deserialize<List<string>>(v, null),
+                        new ValueComparer<ICollection<string>>(
+                            (c1, c2) => c1.SequenceEqual(c2),
+                            c => c.Aggregate(0, (a, v) => HashCode.Combine(a, v.GetHashCode())),
+                            c => (ICollection<string>)c.ToList()));
+            }
 ```
 
-The same thing can be achieved by explicitly specifying the column type. For example, if the entity type is defined like so:
+<!--
+                modelBuilder.Entity<Post>()
+                    .Property(e => e.Tags)
+                    .HasConversion(
+                        v => JsonSerializer.Serialize(v, null),
+                        v => JsonSerializer.Deserialize<List<string>>(v, null),
+                        new ValueComparer<ICollection<string>>(
+                            (c1, c2) => c1.SequenceEqual(c2),
+                            c => c.Aggregate(0, (a, v) => HashCode.Combine(a, v.GetHashCode())),
+                            c => (ICollection<string>)c.ToList()));
+-->
+[!code-csharp[ConfigurePrimitiveCollection](../../../samples/core/Modeling/ValueConversions/PrimitiveCollection.cs?name=ConfigurePrimitiveCollection)]
+
+`ICollection<string>` represents a mutable reference type. This means that a <xref:Microsoft.EntityFrameworkCore.ChangeTracking.ValueComparer%601> is needed so that EF Core can track and detect changes correctly. See [Value Comparers](xref:core/modeling/value-comparers) for more information.
+
+### Collections of value objects
```

Comment:
This specific example seems like it's giving less new info than the others (just showing another example with principles that were already shown above)... Concision is also a virtue (but am also OK with leaving as-is).

Reply:
I wanted to give an example which was a bit less trivial than the others and used nested types. The new concept here is that value objects can be easily composed. This doesn't require making them owned types, for example.

Reply:
Make sense!